src/unread/unreadPmsReducers.js

```diff
@@ -19,10 +19,21 @@ export default (state: UnreadState = initialState, action: Action): UnreadState
     case ACCOUNT_SWITCH:
       return initialState;
 
-    case EVENT_NEW_MESSAGE:
-      return action.message.type === 'private' && action.message.display_recipient.length === 2
-        ? addItemsToPmArray(state, [action.message.id], action.message.sender_id)
-        : state;
+    case EVENT_NEW_MESSAGE: {
+      if (action.message.type !== 'private') {
+        return state;
+      }
+
+      if (action.ownEmail && action.ownEmail === action.message.sender_email) {
+        return state;
+      }
```

Comment:
So, this isn't quite the correct algorithm.  The server actually makes a determination for whether messages sent by you are "already read", and will set message.flags.read if so.  You should check that instead of checking the email address; that's cleaner

As a sidenote, we should always do this sort of comparison by User ID, not email; since User IDs don't change and emails do.  It was a huge pain to fix this category of issue in the webapp to support users changing their email address; might as well avoid adding more checks on email addresses in the mobile app :).